src/getDomainURL.js

```diff
@@ -0,0 +1,27 @@
+import getAPIDomain from "./getAPIDomain";
+
+/**
+ * getDomainURL
+ * @since 2.4.5
+ * @category URL
+ * @description Creates the domain URL using a custom protocol.
+ * @param {Object} options - Options for the function.
+ * @param {boolean} [options.protocol=process.env.REACT_APP_PROTOCOL] - The protocol value to set. Defaults to `process.env.REACT_APP_PROTOCOL`.
+ * @param {Function} options.host - The URL host.
+ * @param {number} options.prefix - The desired prefix for the domain URL.
+ * @returns {string} Returns the domain URL.
+ * @example
+ * getAPIDomain({protocol: 'http', host: 'admin.sweepr.com', prefix: 'api'});
+ * #=> 'http://api.sweepr.com'
+ *
+ * getAPIDomain({protocol: 'https', host: 'admin.sweepr.com', prefix: 'api'});
+ * #=> 'https://api.sweepr.com'
+ */
+export const getDomainURL = ({
+  protocol = process.env.REACT_APP_PROTOCOL,
```

Comment:
We can not expect the consumer of this library to be a React environment, I would rather have the `protocol` field here a required field. If when calling this fn we want a backup, then we can supply it like so:

```
protocol ||= process.env.REACT_APP_PROTOCOL;
const myURL = getDomainURL({protocol, host, prefix, dev});
```

Reply:
Well, we're not building a library that can be reused by everyone, everywhere, right? It's just a utility for our use cases, where the primary one is a React app. Plus, if that's going to be a required field, then there is no point in providing a fallback. The function should just throw an error.

As it is right now, it does the same thing as in your example, but it's pretty explicit if you check the JSDoc. You've got the details about the default variable mentioned there.
<img width="686" alt="image" src="https://github.com/rosebud-iot/ui-utils/assets/105205711/e69401bf-d67e-4326-a770-10c86d08f073">

If we would supply the backup inside the function body, you wouldn't know what's happening there unless you check the code.

If we don't want the fallback I suppose that it's better to remove it entirely.

Reply:
> Well, we're not building a library that can be reused by everyone, everywhere, right?

We're trying to keep it as generic as possible. UI-Utils is imported on ui-consumer repo and ui-admin right now, possibly will expand that in the future as we build more side projects.

The utilities are sometimes quite specific to Sweepr's needs right now, but if too specific to a certain repository/stack, we keep those utilities in that repository. Like [ui-admin's INT utils](https://github.com/rosebud-iot/ui-admin/blob/f0d5452b3460ead37fbe269efc4faeac8408b839/src/apps/INT/lib/utils.js) - wouldn't be useful outside of the experience builder.